### ISSUE TITLE
proposal: the docker client should load certs from certs.d in addition to ~/.docker/tls

### ISSUE THREAD

**Issue by vikstrous** [docker/docker#26644]
When #12756 is solved and certs.d becomes more commonly used to trust private registries, the next issue that users will run into is that the docker client and the daemon have different CA directories. This is an issue when trying to pull with content trust enabled and you have a private registry cert only in one place, but not the other.

I'm proposing making the client trust all CA certs from certs.d in addition to the current `~/.docker/tls/example.com/`. This takes care of the common case of having both on the same machine.

I'm also interested in how to make this experience better in the case where the client and server are not on the same machine, but that would be a much more controversial topic and I don't know which direction the trust should go in. If anyone wants to discuss that, please open a separate issue and keep this one for the above proposal.

**Comment by justincormack** [comment]
For Docker for Mac and Windows we are planning to trust all the certs on the client machine for the daemon. This is a slightly unusual case as these are on the same laptop I guess. But I am not sure why you wouldnt just put CA certs in the system CA cert directories, in which case they would work for both anyway?

**Comment by vikstrous** [comment]
Yeah, trusting the certs globally is generally the most reliable option. However, when users/customers have a large number of different distros and each of them has their own way of managing CA certs, it would be much easier to be able to tell them "just put the cert in X directory" instead of "look up how to add certs globally for your distro".
